raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/RawSpansGrouper.java

```diff
@@ -75,7 +89,34 @@ public StreamsBuilder buildTopology(
     streamsBuilder.addStateStore(spanStoreBuilder);
     streamsBuilder.addStateStore(traceStateStoreBuilder);
 
-    StreamPartitioner<TraceIdentity, StructuredTrace> groupPartitioner =
+    KStream<byte[], PreProcessedSpan> preProcessedStream =
+        inputStream.transform(() -> new JaegerSpanPreProcessor(getGrpcChannelRegistry()));
```

Comment:
How are we ensuring that this jaeger spans topic is consumed exactly from the offset where the span-normalizer got decommissioned?

Reply:
Good point. Being different applications, it will be difficult to get them in sync.

Reply:
My suggestion would be the following:
We shouldn't even be reading from the existing jaeger-spans topic. This topology should read from a different topic(say jaeger-spans-v2). On the producer side(i.e. hypertrace-collector), we should have a way to control whether to produce to the existing jaeger-spans topic or to the new jaegar-spans-v2. Once we make the flip to the v2 topic we can decommission span-normalizer.

Reply:
Probably need to keep both applications for now. And may need to move `raw-span-grouper` logic to `span-normalizer` instead of other way around.